yt_dlp/extractor/adn.py

```diff
@@ -182,10 +182,15 @@ def _real_extract(self, url):
             'Downloading player config JSON metadata',
             headers=self._HEADERS)['player']
         options = player['options']
+        video = options['video']
+        startDate = video.get('startDate')
+        currentDate = video.get('currentDate')
+        if startDate and currentDate and startDate > currentDate:
+            raise ExtractorError(f'This video is not available yet. Release date: {startDate}', expected=True)
```

Comment:
Are these values unix timestamps? And if so, should we rely on `currentDate` instead of `time.time()`? IMO we should do this a bit more safely (and use snake_case):
```suggestion
        start_date = traverse_obj(options, ('video', 'startDate', {int}))
        current_date = traverse_obj(options, ('video', 'startDate', {int}))
        if start_date and current_date and start_date > current_date:
            raise ExtractorError(f'This video is not available yet. Release date: {start_date}', expected=True)
```
and if these are unix timestamps, `Release date: {start_date}` is not useful IMO

Perhaps we should even check this after the `if not formats:` statement below?

Reply:
These are ISO 8601 UTC timestamps. Do I need to convert that to unix timestamps or `datetime` for comparison?
If this is a future video we won't reach the formats section, because `user.get('hasAccess')` will be false.
Thanks for pointing out snake case, that's what you get for relying on Copilot too much 🤦
I'll push a fix tomorrow.

Reply:
ah ok. I'd import `parse_iso8601` from utils and do this then:
```suggestion
        start_date = traverse_obj(options, ('video', 'startDate', {str}))
        if (parse_iso8601(start_date) or 0) > time.time():
            raise ExtractorError(f'This video is not available yet. Release date: {start_date}', expected=True)
```